### PR TITLE
Enable more ESLint rules

### PR DESCRIPTION
**Pull request by Turbo87**
No ES6 stuff yet, just general cleanup.

### CODE REVIEW

**Comment by Turbo87** [comment]
@ember-cli/core comments please :)

File: .eslintrc.js [hunk 1]
```diff
@@ -10,113 +13,84 @@ module.exports = {
   rules: {
```
Comment (hjdivad) [inline .eslintrc.js]: should we add [dot-location](http://eslint.org/docs/rules/dot-location)?

**Comment by hjdivad** [comment]
> I reviewed the rule changes in .eslintrc.js and they look good to me, but I have not reviewed the full diff.

I reviewed the same as @rwjblue.  I could almost 👍  his comment except I'm enthusiastic about trailing commas ^_^

**Comment by Turbo87** [comment]
@hjdivad since `dot-location` can be controversial I left it out of the PR. we should probably discuss it outside of this PR and then open another one if we feel like we should add it.

**Comment by hjdivad** [comment]
> @hjdivad since dot-location can be controversial I left it out of the PR. we should probably discuss it outside of this PR and then open another one if we feel like we should add it.

My thinking was that while dot-location can be controversial, `consistency > mixed` would be uncontroversial even for consistency of someone's unpreferred approach.

That said, deferring to a follow up PR sounds sensible to me 👍

**Comment by rwjblue** [comment]
> My thinking was that while dot-location can be controversial, consistency > mixed would be uncontroversial even for consistency of someone's unpreferred approach.

Agreed. I could definitely live with either, but prefer `property` personally...

**Comment by Turbo87** [comment]
Rebased and fixed the conflicts. I'll merge once CI passes.

**Comment by Turbo87** [comment]
actually... @homu r+

**Pull request merged**